src/commands/register.ts

```diff
@@ -2,142 +2,161 @@ import { flags } from '@oclif/command';
 import chalk from 'chalk';
 import { classToPlain } from 'class-transformer';
 import { cli } from 'cli-ux';
-import inquirer from 'inquirer';
+import fs from 'fs-extra';
 import path from 'path';
 import untildify from 'untildify';
 import Command from '../base-command';
 import LocalDependencyManager from '../common/dependency-manager/local-manager';
 import MissingContextError from '../common/errors/missing-build-context';
 import { buildImage, getDigest, pushImage, strip_tag_from_image } from '../common/utils/docker';
-import { ServiceNode } from '../dependency-manager/src';
+import { ServiceConfig, ServiceNode } from '../dependency-manager/src';
+import { ComponentConfigBuilder } from '../dependency-manager/src/component-config/builder';
 
-export interface CreateServiceVersionInput {
+export interface PutComponentVersionDto {
   tag: string;
-  digest: string;
   config: any;
 }
 
-export default class ServiceRegister extends Command {
-  static aliases = ['service:register', 'services:register', 'svcs:register', 'svc:register'];
-  static description = 'Register a new Service with Architect Cloud';
+export default class ComponentRegister extends Command {
+  static aliases = ['component:register', 'components:register', 'c:register', 'comp:register'];
+  static description = 'Register a new Component with Architect Cloud';
 
   static flags = {
     ...Command.flags,
-    services: flags.string({
-      char: 's',
-      description: 'Path to a service to build',
+    components: flags.string({
```

Comment:
Should this be singular?

Reply:
I thought the same thing, but previously `services` was plural so I kept it the same.

Paging @davidthor for opinion:

```bash
architect register --components ./architect.yml
## OR
architect register --component architect.yml

## ...keeping in mind that it accepts multiple paths:
architect register --components ./architect.yml ../another/architect.yml
```

Reply:
I think the `--components` flag is going to be exponentially less useful with the current design. Related services are likely to be in the same component making it unlikely you'd be publishing multiple components at once.